src/pages/tests/CreateEditRecipePage.test.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable testing-library/no-wait-for-multiple-assertions */
+import { render, screen, waitFor, act } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import CreateEditRecipePage from "../CreateEditRecipePage";
+import { Router, Route } from "react-router-dom";
+import { createMemoryHistory } from "history";
+
+const history = createMemoryHistory();
+const renderPage = (path) =>
+  render(
+    <Router history={history}>
+      <Route path={path}>
+        <CreateEditRecipePage />
+      </Route>
+    </Router>
+  );
+
+describe("Create recipe", () => {
+  const renderCreatePage = () => {
+    history.push("/recipes/new");
+    renderPage("/recipes/new");
+  };
+
+  it("loads create recipe form", () => {
+    renderCreatePage();
+
+    screen.getByRole("heading", { name: /Create new recipe/i });
+
+    const nameInput = screen.getByLabelText(/name/i);
```

Comment:
you should be able to add a `data-testid` to the elements that you want to test it and use it in here.
For example:
`<InputField type="text" id="name" ........ data-testid="input-recipe-name"/>`

Then in the test you can do:
`const nameInput = screen.getByTestId("input-recipe-name")`